pkg/crc/preflight/preflight_check_tray_darwin.go

```diff
@@ -35,13 +35,22 @@ type TrayVersion struct {
 }
 
 func checkIfDaemonPlistFileExists() error {
-	if !launchd.PlistExists(daemonAgentLabel) {
-		return fmt.Errorf("Daemon plist file does not exist")
-	}
-	return nil
+	// crc setup can be ran from any location in the
+	// users computer, and we need to update the plist
+	// file with the path of the crc binary which was
+	// used to run setup, to force it this check needs
+	// to always fail so the fix routine is triggered
+
+	return fmt.Errorf("Daemon plist file does not exist")
```

Comment:
If this check always going to return the `error` then what is the use of this check we should just remove it.

Reply:
Ohh, that'd be nice, this is just there to trigger the fix, can we make the fix run without having the check..?

Reply:
@anjannath why not, may be you can change the name without `fix` prefix and run it for every single time?

Reply:
@praveenkumar In that way we have to completely move this out of the preflight checks, now we need this to fill the `preflightCheck` struct, if we leave out the check member in that struct, i think we'll hit a segfault.

Reply:
I don't know if it's expensive to recreate this plist file, or if it has side effects we want to avoid, but an alternative could be to change `checkIfDaemonPlistFileExists` to 
- generate a string containing to the content the plist file should have for the current binary
- compare that string with the content of the plist file we have on disk
- only return an error if this does not match
If recreating the plist file is cheap enough, then maybe this is overkill..

Reply:
i didn't notice an increase in time consumed, so its very cheap to create it, happens near instantly

Reply:
it's not just file creation, but file creation + LoadPlist + RestartAgent which I was asking if there are noticeable side effects.

Reply:
oh, I did not notice any side effects, seems to work as expected

Reply:
@cfergeau @anjannath I would still think if we are going to keep it then better to update the error message because  `Daemon plist file does not exist` seems confusing to me, what about just put `Ignoring this check` as error message and explanation already part of comments?